tests/ga/utils/test_exthandler_utils.py

```diff
@@ -0,0 +1,1252 @@
+    DEFAULT_MEM_LIMIT_MIN_MB_FOR_EXTN
+from tests.ga.test_extension import ExtensionTestCase
+from tests.tools import AgentTestCase, load_data_path
+
+
+@patch("azurelinuxagent.common.osutil.default.DefaultOSUtil.get_total_mem", return_value=1024)
+class TestCGroupsLimits(AgentTestCase):
+    def test_no_limits_passed(self, patch_get_total_mem):
+        cgroup_name = "test_cgroup"
+        limits = CGroupsLimits(cgroup_name)
+        self.assertEqual(limits.cpu_limit, CGroupsLimits.get_default_cpu_limits(cgroup_name))
+        self.assertEqual(limits.memory_limit, CGroupsLimits.get_default_memory_limits(cgroup_name))
+
+        limits = CGroupsLimits(None)
+        self.assertEqual(limits.cpu_limit, CGroupsLimits.get_default_cpu_limits(cgroup_name))
+        self.assertEqual(limits.memory_limit, CGroupsLimits.get_default_memory_limits(cgroup_name))
+
+    def test_agent_name_limits_passed(self, patch_get_total_mem):
+        cgroup_name = AGENT_CGROUP_NAME
+        limits = CGroupsLimits(cgroup_name)
+        self.assertEqual(limits.cpu_limit, CGroupsLimits.get_default_cpu_limits(cgroup_name))
+        self.assertEqual(limits.memory_limit, CGroupsLimits.get_default_memory_limits(cgroup_name))
+
+        data = '''{
+                  "name": "ExampleHandlerLinux",
+                  "version": 1.0,
+                  "handlerConfiguration": {
+                    "linux": {
+                      "resources": {
+                        "cpu": [
+                          {
+                            "cores": 2,
+                            "limit_percentage": 25
+                          },
+                          {
+                            "cores": 8,
+                            "limit_percentage": 20
+                          },
+                          {
+                            "cores": -1,
+                            "limit_percentage": 15
+                          }
+                        ],
+                        "memory": {
+                          "max_limit_percentage": 20,
+                          "max_limit_MBs": 1000,
+                          "memory_pressure_warning": "low",
+                          "memory_oom_kill": "enabled"
+                        }
+                      }
+                    },
+                    "windows": {}
+                  }
+                }
+                '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        resource_config = handler_config.get_resource_configurations()
+
+        with patch(
+                "azurelinuxagent.common.osutil.default.DefaultOSUtil.get_processor_cores") as patch_get_processor_cores:
+            total_ram = 1024
+            patch_get_processor_cores.return_value = 8
+            patch_get_total_mem.return_value = total_ram
+
+            expected_cpu_limit = 20
+            expected_memory_limit = 256  # .2 of total ram is lower than default, reset to default.
```

Comment:
Yes. This "DEFAULT" value is kept in there with the variety of VMs in mind. This would be unnecessary with the `min_limit_MBs` flag in the future, where the onus comes on the extensions to get the minimum value set and thus would make DEFAULTs unnecessary in those cases.